llm/memory.go

```diff
@@ -440,3 +448,39 @@ func projectorMemoryRequirements(filename string) (weights, graphSize uint64) {
 
 	return weights, graphSize
 }
+
+// estimateKvCacheSize determines the memory required for K or V cache based on the quantization type
+func estimateKvCacheSize(cacheType string, numCtx, blockCount, embeddingHeadCount, headCountKV uint64, isEmbeddingModel bool) uint64 {
+	var bytesPerElement float64
+
+	if isEmbeddingModel && cacheType != "f16" && cacheType != "f32" {
+		cacheType = "f16" // Default to f16 for embedding models if an unsupported type is specified
+	}
+
+	switch cacheType {
+	case "f32", "fp32":
+		bytesPerElement = 4 // fp32
+	case "", "f16", "fp16":
+		bytesPerElement = 2 // fp16
+	case "q8_0":
+		bytesPerElement = 1 // 1/2 of fp16
+	case "q5_1":
```

Comment:
Note: As I'm now mirroring the llama.cpp implementation for GGML type to quantisation format I've included the q5_1,q5_0,iq4_nl,q4_1 options as they're what's standardised on supporting upstream - and q5_1 might actually be a nice saving on q8_0 without dropping to q4_0. In the FAQ docs I'm still calling out q8_0 and q4_0 as being the two most popular quantised options and recommending q8_0 as a good starting point.